src/hash.rs

```diff
@@ -140,4 +150,14 @@ where
         self.write(right.as_ref());
         self.hash()
     }
+
+    /// Returns hash value for MT interior node (prefix 0x01).
+    #[inline]
+    fn multi_node(&mut self, nodes: &[T], _height: usize) -> T {
+        self.write(&[INTERIOR]);
```

Comment:
This is for compatibility.  The tagging does exist in merkle_light and has since I've taken it up.  It can be removed, but just know it wasn't introduced as new behaviour with this code for any reason.

Reply:
It should stay here, merkle trees should be tagged by default. We overwrite these methods in filecoin anyway

Reply:
^ That was my understanding as well.

Reply:
@cryptonemo Yeah, that's what I meant by legacy. I personally think any such legacy support should be passive (i.e. because it's convenient) rather than actively supported as a matter of principle. I think we'll end up with the best long-term outcomes if the library defaults to handling our needs and supports others as special cases where unproblematic. Here I'm addressing @dignifiedquire's statement of what defaults 'should' be. That said, I can disagree and commit.

Reply:
I like security first, by default ;)